src/System/Random.hs

```diff
@@ -197,9 +200,15 @@ class Random a where
   -- | Takes a range /(lo,hi)/ and a pseudo-random number generator
   -- /g/, and returns a pseudo-random value uniformly distributed over the
```

Comment:
We should not promise that it is __uniformly__ distributed.

Reply:
I do agree that current wording isn't perfect, but that's a tricky one because it is uniform for most of the types. Maybe we can say instead:

> and for types that have a well defined range concept it returns a pseudo-random value uniformly distributed over the closed interval /[lo,hi]/...

Otherwise saying "closed interval" is not terribly accurate either. What does it mean for `(a, b)` to be inclusive, right?

Types that behave differently from "uniformly" are now documented, so it is less of a problem if we get it worded properly.

Reply:
I would probably write "For a tuple `(lo, hi)` and a pseudo-random number generator `g`, return a pseudo-random value between `lo` and `hi` (inclusive) together with a new generator".

Reply:
Sorry for sidetracking and not clarifying myself. I only said

> What does it mean for (a, b) to be inclusive, right?

as a way to argue a point and it was more of a rhetorical question.

To be more precise I meant `rangeR ((alo, blo), (ahi, bhi)) g :: ((a, b), g)`. Clusivity with instances as they are defined in this PR only makes sense if `a` and `b` types are kept independent of each other.

Reply:
I think problem here is not "uniform", proposed randomR for tuples generate uniform distribution over N-dimensional rectangle. It conflict with "over the closed interval /[lo,hi]/". 

This is really question what is `Random`? What does it promise? Should `randomR` generate values in the *interval*? That will rule out instances for tuples. Should it generate values in some set specified by tuple (a,b) which exact meaning depends on type in question?

Reply:
> This is really question what is Random? What does it promise?

The promise is that the values a random. Everything else is open to an interpretation on per type basis. 

> Should it generate values in some set specified by tuple (a,b) which exact meaning depends on type in question?

I think it is a reasonable assumption. A subset of possible values that is defined by a tuple in some sensible way. With exact documentation what that sensible way is.

Reply:
I don't think these instances are controversial in any way, moreover most programmers will likely infer exactly what those instances do without any more explanation. Resolving this discussion, anyone seeking an improvement to haddock can add a PR later on.